### ISSUE TITLE
Upgrade to 7.1.1. From 6.67.0 fails due to assertion failure

### ISSUE THREAD

**Issue by sunilrr** [pulumi/pulumi-gcp#1363]
### What happened?

Secret prior to v7,
```
replication=gcp.secretmanager.SecretReplicationArgs(
                automatic=True,
            ),
```

After upgrade to v7,
```
replication=gcp.secretmanager.SecretReplicationArgs(
                auto=gcp.secretmanager.SecretReplicationAutoArgs(),
            ),
```

preview on existing stack fails with error,
```
gcp:secretmanager:Secret some-secret update [diff: ~provider] 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running panic: fatal: An assertion has failed: Expected diff to not require deletion or replacement during Update of urn:pulumi:test-stack::pulumi-gcp::flm:templates:abc$gcp:secretmanager/secret:Secret::some-secret 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running goroutine 362 [running]: 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.93.0/go/common/util/contract/failfast.go:23 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assertf(0xe0?, {0x4e6dfd3?, 0x0?}, {0xc007807610?, 0xc006464600?, 0x58631a0?}) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.93.0/go/common/util/contract/assert.go:35 +0xe8 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Update(0xc0006a1340, {0x585b4c0?, 0xc006464570?}, 0xc0005293b0) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.65.0/pkg/tfbridge/provider.go:980 +0x765 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Update.func1({0x5875ec8?, 0xc0006a1340?}) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:356 +0x33 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi-terraform-bridge/x/muxer.resourceMethod[...](0xc00267b310?, 0x40, 0xc007807760?) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:303 +0xb5 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Update(0x0?, {0x585b4c0?, 0xc006464570?}, 0x40?) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.7-0.20230801203955-5d215c892096/muxer.go:355 +0x65 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Update_Handler.func1({0x585b4c0, 0xc006464570}, {0x4b2bb60?, 0xc0005293b0}) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.93.0/proto/go/provider_grpc.pb.go:629 +0x75 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x585b4c0, 0xc006464240}, {0x4b2bb60, 0xc0005293b0}, 0xc0063b3ca0, 0xc007dcdaa0) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3d0 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Update_Handler({0x4c67200?, 0xc00267b310}, {0x585b4c0, 0xc006464240}, 0xc0063c4af0, 0xc0007f8460) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.93.0/proto/go/provider_grpc.pb.go:631 +0x135 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001281e0, {0x586ee40, 0xc005c74000}, 0xc0064147e0, 0xc006a41530, 0x968be10, 0x0) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running google.golang.org/grpc.(*Server).handleStream(0xc0001281e0, {0x586ee40, 0xc005c74000}, 0xc0064147e0, 0x0) 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running google.golang.org/grpc.(*Server).serveStreams.func1.1() 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:986 +0xbb 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 37 
November 18th 2023 10:45:07
Info
    pulumi:pulumi:Stack test-stack running 	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.58.3/server.go:997 +0x145 
```

### Example

```
replication=gcp.secretmanager.SecretReplicationArgs(
                auto=gcp.secretmanager.SecretReplicationAutoArgs(),
            ),
```

### Output of `pulumi about`

`version = "7.1.1"`

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by mikhailshilkov** [comment]
I'm able to repro this.

Deploy in v6 (I had 6.67.0):

```
import * as pulumi from '@pulumi/pulumi';
import * as gcp from '@pulumi/gcp';

const replication= new gcp.secretmanager.Secret("test", {
  secretId: "test-mikhail",
  replication: {
    automatic: true,
  },
});
```

then upgrade the provider to 7.2.0 and change the code to:

```
import * as gcp from '@pulumi/gcp';

const replication= new gcp.secretmanager.Secret("test", {
  secretId: "test-mikhail",
  replication: {
    auto: {},
  },
});
```

Then `pulumi preview` fails with a panic.

**Comment by iwahbe** [comment]
I'm triggering the patch release now. This should be fixed in `v7.2.1`.

**Comment by iwahbe** [comment]
I believe this was fixed in https://github.com/pulumi/pulumi-gcp/releases/tag/v7.2.1. Please let us know if this issue persists for you.
